draft-gpew-priv-ppm.md

```diff
@@ -713,25 +719,23 @@ the first aggregate request. Shall we exclude it in subsequent requests
 somehow?] The remainder of the structure is dedicated to VDAF-specific request
 parameters.
 
-In order to provide replay protection, the leader is required to send aggregate
-sub-requests in ascending order, where the ordering on sub-requests is
-determined by the algorithm defined in {{anti-replay}}. Specifically, the leader
-constructs its request such that:
-
-* each sub-request follows the previous sub-request; and
-* the first sub-request follows the last sub-request in the previous aggregate
-  request.
+In order to provide replay protection, the leader preprocesses the set of
+reports it sends in the the AggregateReq as described in {{anti-replay}}. Any
+reports filtered out by this procedure MUST be ignored.
 
 The helper handles well-formed requests as follows. (As usual, malformed
 requests are handled as described in {{errors}}.) It first looks for PPM
-parameters corresponding to `AggregateReq.task_id`. It then filters out
-out-of-order sub-requests by ignoring any sub-request that does not follow the
-previous one (See {{anti-replay}}.)
+parameters corresponding to `AggregateReq.task_id`. It then preprocesses the
+sub-requests as described in {{anti-replay}}. Any sub-requests filtered out by
+this procedure MUST be ignored.
+
+In addition, the helper MUST ignore any report whose nonce contains a timestamp
+that falls in a batch interval for which it has received at least one
```

Comment:
This also is a recipe for silent error.

Reply:
Yeah I think you're right that failing silently here is a bad idea. There are other ways in which the helper might fail for a given report, and all of these should be handled by relaying an error back to the leader. I have an idea for how to address this in a way that handles all such errors in a uniform manner (see #179 if you're curious), but it would be hard to specify very concretely given the state the document is in. I've updated the text to make sure the helper sends an error here. It's a bit awkward, but it'll do the job for now.

Reply:
Totally agree here. I've added some text saying that the helper MUST reply with an error in this case rather than its next VDAF message. The text is a bit awkward, but I think it's hard to do much better in the current state of the document. In fact, there are a number of errors that might happen between now and when the helper is ready to reply. In #179 I've tried to solve this problem in a more generic way.